docker-compose.yml

```diff
@@ -1,88 +1,120 @@
 # Some people have reported problems with the balrogui API proxy not working correctly.
 # This appears to be the same problem reported in https://github.com/docker/compose/issues/2172,
 # and seems to only occur with certain (older) versions of docker-compose.
-balrogadmin:
-  build: .
-  dockerfile: Dockerfile.dev
-  entrypoint:
-    - bash
-    - scripts/initdb_and_run.sh
-  command: admin-dev
-  expose:
-    - "7070"
-  volumes:
-    - .:/app
-  links:
-    - balrogdb
-  environment:
-    - DBURI=mysql://balrogadmin:balrogadmin@balrogdb/balrog
-    - SECRET_KEY=blahblah
-    - PORT=7070
-    - LOG_FORMAT=plain
-    - LOG_LEVEL=WARNING
-    # Grab mail information from the local environment
-    - SMTP_HOST
-    - SMTP_PORT
-    - SMTP_USERNAME
-    - SMTP_PASSWORD
-    - SMTP_TLS
-    - NOTIFY_TO_ADDR
-    - NOTIFY_FROM_ADDR
+version: '2.1'
+services:
+  balrogadmin:
+    build:
+      context: .
+      dockerfile: Dockerfile.dev
+    depends_on:
+      balrogdb:
+        condition: service_healthy
+    entrypoint:
+      - bash
+      - scripts/initdb_and_run.sh
+    command: admin-dev
+    expose:
+      - "7070"
+    volumes:
+      - .:/app
+    links:
+      - balrogdb
+    environment:
+      - DBURI=mysql://balrogadmin:balrogadmin@balrogdb/balrog
+      - SECRET_KEY=blahblah
+      - PORT=7070
+      - LOG_FORMAT=plain
+      - LOG_LEVEL=WARNING
+      # Grab mail information from the local environment
+      - SMTP_HOST
+      - SMTP_PORT
+      - SMTP_USERNAME
+      - SMTP_PASSWORD
+      - SMTP_TLS
+      - NOTIFY_TO_ADDR
+      - NOTIFY_FROM_ADDR
+    healthcheck:
+        test: nc -z -v balrogadmin 7070
+        interval: 5s
+        timeout: 30s
+        retries: 3
```

Comment:
This seems to fail sometimes, at least on my machine (and I suspect it would fail more often on slower machines). It's probably best to use a high number of retries (at least 10). Same thing goes for the db healthcheck.